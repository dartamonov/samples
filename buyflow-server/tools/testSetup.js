// Tests are placed alongside files under test.
// This file does the following:
// 1. Sets the environment to 'production' so that
//    dev-specific babel config in .babelrc doesn't run.
// 2. Disables Webpack-specific features that Mocha doesn't understand.
// 3. Registers babel for transpiling our code for testing.

// This assures the .babelrc dev config (which includes
// hot module reloading code) doesn't apply for tests.
// process.env.NODE_ENV = 'production';
// process.env.NODE_ENV = 'test';

// Register babel so that it will transpile ES6 to ES5
// before our tests run.
require('babel-register')();
//require("babel-polyfill");
require("babel-jest");
