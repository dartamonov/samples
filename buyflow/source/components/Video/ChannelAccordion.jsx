import React, { PropTypes, Component } from 'react';
import { Row, Col } from '../Grid';
import Accordion from '../Accordion/Accordion.jsx';
import ChannelOption from './ChannelOption.jsx';

class ChannelAccordion extends Component {

  handleAccordionClick(expanded) {
    const {
      toggleAccordionAction,
      channelGroup
    } = this.props;

    toggleAccordionAction(channelGroup.channelType, !expanded);
  }

  render() {
    const {
      channelGroup,
      updateChannelSelectionAction,
      accordion
    } = this.props;

    return (
      <Accordion active={channelGroup.expanded || !accordion}>
        {accordion ? <div className="hg-panel hg-panel--action-right-at-768 accordion__title">
          <div className="hg-panel__content">
            <div className="hgroup">
              <h3>{channelGroup.title}</h3>
            </div>
          </div>
          <div className="hg-panel__action">
            <button data-tracking={{}} type="button" onClick={this.handleAccordionClick.bind(this, channelGroup.expanded)} className="x-body3 x-button--text accordion__trigger" aria-expanded={`${channelGroup.expanded}`}>
              {!channelGroup.expanded
                ? <span>Show Channels</span>
                : <span>Hide Channels</span>
              }
            </button>
          </div>
        </div> : null}
        <div className="accordion__body accordion__body--no-panel margin-top24">
          {!accordion &&
            <h3 className="x-heading1">{channelGroup.title}</h3>
          }
          <Row>
            {channelGroup.channels.map((option, index) => {
                return (
                  <Col className="margin-top12" md={6} xs={12} key={index}>
                    <ChannelOption
                      index={index}
                      option={option}
                      channelGroup={channelGroup}
                      updateChannelSelectionAction={updateChannelSelectionAction}
                    />
                  </Col>
                );
              })
            }
          </Row>
        </div>
      </Accordion>
    );
  }
}

ChannelAccordion.propTypes = {
  channelGroup: PropTypes.object,
  accordion: PropTypes.boolean,
  updateChannelSelectionAction: PropTypes.func,
  toggleAccordionAction: PropTypes.func
};

export default ChannelAccordion;
