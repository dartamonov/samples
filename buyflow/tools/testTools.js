import renderer from 'react-test-renderer';

export const takeSnapshot = x => renderer.create(x).toJSON();