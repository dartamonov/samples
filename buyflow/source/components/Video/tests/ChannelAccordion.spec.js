import React from 'react';
import ChannelAccordion from '../ChannelAccordion.jsx';
import {takeSnapshot} from '../../../tools/testTools';

describe('ChannelAccordion component', () => {
  it('handles accordion with a header', () => {
    const accordion = true;
    const channelGroup = {
      "channels": [
        {
          "logoUrl": "logoUrl.png",
          "backgroundImageUrl": "backgroundImageUrl.png",
          "id": "610379779",
          "name": "Starz",
          "description": "Starz®, Starz Edge™, Starz Cinema®, Starz Kids & Family® & more",
          "price": 10.99,
          "isSelected": true
        },
        {
          "logoUrl": "logoUrl.png",
          "backgroundImageUrl": "backgroundImageUrl.png",
          "id": "610371595",
          "name": "HBO",
          "description": "HBO®, HBO 2®, HBO Comedy®, HBO Latino®, HBO Family & more",
          "price": 19.99,
          "isSelected": false
        }
      ],
      "channelType": "VideoPremiumChannel",
      "weight": 0,
      "isModuleExpanded": true,
      "title": "Pick channels to order"
    };
    const snapshot = takeSnapshot(
      <ChannelAccordion
        channelGroup={channelGroup}
        accordion={accordion}
      />
    );

    expect(snapshot).toMatchSnapshot();
  });
});
