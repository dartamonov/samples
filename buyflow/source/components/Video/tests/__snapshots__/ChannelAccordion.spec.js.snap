// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart component handles accordion with a header 1`] = `
<div
  aria-expanded={false}
  className="accordion"
>
  <div
    className="hg-panel hg-panel--action-right-at-768 accordion__title"
  >
    <div
      className="hg-panel__content"
    >
      <div
        className="hgroup"
      >
        <h3>
          Pick channels to order
        </h3>
      </div>
    </div>
    <div
      className="hg-panel__action"
    >
      <button
        aria-expanded="undefined"
        className="x-body3 x-button--text accordion__trigger"
        data-tracking={Object {}}
        onClick={[Function]}
        type="button"
      >
        <span>
          Show Channels
        </span>
      </button>
    </div>
  </div>
  <div
    className="accordion__body accordion__body--no-panel margin-top24"
  >
    <div
      className="row"
    >
      <div
        className="margin-top12 col-md-6 col-xs-12"
      >
        <div
          className="customize-channel-panel"
          style={
            Object {
              "backgroundImage": "url(backgroundImageUrl.png)",
            }
          }
        >
          <div
            className="customize-channel-panel__content"
          >
            <div
              className="customize-channel-panel__content--top"
            >
              <span
                className="helper"
              />
              <img
                alt="Starz Logo"
                src="logoUrl.png"
              />
            </div>
            <div
              className="customize-channel-panel__content--bottom"
            >
              <p>
                <span
                  className="x-body3"
                >
                  Starz®, Starz Edge™, Starz Cinema®, Starz Kids & Family® & more
                </span>
              </p>
            </div>
          </div>
          <div
            className="customize-channel-panel__input"
          >
            <div
              className="x-field--checkbox"
            >
              <div>
                <label
                  className="x-checkbox-label x-body2"
                  htmlFor="610379779"
                >
                  <span
                    className="vh"
                  >
                    Starz
                     
                  </span>
                  <span
                    className="hg-price"
                  >
                    <span
                      className="hg-price__icon"
                    >
                      $
                    </span>
                    <span
                      className="hg-price__dollars"
                    >
                      10
                    </span>
                    <span
                      className="hg-price__cents"
                    >
                      .99
                    </span>
                    <span>
                      /mo
                    </span>
                  </span>
                </label>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="margin-top12 col-md-6 col-xs-12"
      >
        <div
          className="customize-channel-panel"
          style={
            Object {
              "backgroundImage": "url(backgroundImageUrl.png)",
            }
          }
        >
          <div
            className="customize-channel-panel__content"
          >
            <div
              className="customize-channel-panel__content--top"
            >
              <span
                className="helper"
              />
              <img
                alt="HBO Logo"
                src="logoUrl.png"
              />
            </div>
            <div
              className="customize-channel-panel__content--bottom"
            >
              <p>
                <span
                  className="x-body3"
                >
                  HBO®, HBO 2®, HBO Comedy®, HBO Latino®, HBO Family & more
                </span>
              </p>
            </div>
          </div>
          <div
            className="customize-channel-panel__input"
          >
            <div
              className="x-field--checkbox"
            >
              <div>
                <label
                  className="x-checkbox-label x-body2"
                  htmlFor="610371595"
                >
                  <span
                    className="vh"
                  >
                    HBO
                     
                  </span>
                  <span
                    className="hg-price"
                  >
                    <span
                      className="hg-price__icon"
                    >
                      $
                    </span>
                    <span
                      className="hg-price__dollars"
                    >
                      19
                    </span>
                    <span
                      className="hg-price__cents"
                    >
                      .99
                    </span>
                    <span>
                      /mo
                    </span>
                  </span>
                </label>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ChannelAccordion component handles accordion with a header 1`] = `
<div
  aria-expanded={false}
  className="accordion"
>
  <div
    className="hg-panel hg-panel--action-right-at-768 accordion__title"
  >
    <div
      className="hg-panel__content"
    >
      <div
        className="hgroup"
      >
        <h3>
          Pick channels to order
        </h3>
      </div>
    </div>
    <div
      className="hg-panel__action"
    >
      <button
        aria-expanded="undefined"
        className="x-body3 x-button--text accordion__trigger"
        data-tracking={Object {}}
        onClick={[Function]}
        type="button"
      >
        <span>
          Show Channels
        </span>
      </button>
    </div>
  </div>
  <div
    className="accordion__body accordion__body--no-panel margin-top24"
  >
    <div
      className="row"
    >
      <div
        className="margin-top12 col-md-6 col-xs-12"
      >
        <div
          className="customize-channel-panel"
          style={
            Object {
              "backgroundImage": "url(backgroundImageUrl.png)",
            }
          }
        >
          <div
            className="customize-channel-panel__content"
          >
            <div
              className="customize-channel-panel__content--top"
            >
              <span
                className="helper"
              />
              <img
                alt="Starz Logo"
                src="logoUrl.png"
              />
            </div>
            <div
              className="customize-channel-panel__content--bottom"
            >
              <p>
                <span
                  className="x-body3"
                >
                  Starz®, Starz Edge™, Starz Cinema®, Starz Kids & Family® & more
                </span>
              </p>
            </div>
          </div>
          <div
            className="customize-channel-panel__input"
          >
            <div
              className="x-field--checkbox"
            >
              <div>
                <label
                  className="x-checkbox-label x-body2"
                  htmlFor="610379779"
                >
                  <span
                    className="vh"
                  >
                    Starz
                     
                  </span>
                  <span
                    className="hg-price"
                  >
                    <span
                      className="hg-price__icon"
                    >
                      $
                    </span>
                    <span
                      className="hg-price__dollars"
                    >
                      10
                    </span>
                    <span
                      className="hg-price__cents"
                    >
                      .99
                    </span>
                    <span>
                      /mo
                    </span>
                  </span>
                </label>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="margin-top12 col-md-6 col-xs-12"
      >
        <div
          className="customize-channel-panel"
          style={
            Object {
              "backgroundImage": "url(backgroundImageUrl.png)",
            }
          }
        >
          <div
            className="customize-channel-panel__content"
          >
            <div
              className="customize-channel-panel__content--top"
            >
              <span
                className="helper"
              />
              <img
                alt="HBO Logo"
                src="logoUrl.png"
              />
            </div>
            <div
              className="customize-channel-panel__content--bottom"
            >
              <p>
                <span
                  className="x-body3"
                >
                  HBO®, HBO 2®, HBO Comedy®, HBO Latino®, HBO Family & more
                </span>
              </p>
            </div>
          </div>
          <div
            className="customize-channel-panel__input"
          >
            <div
              className="x-field--checkbox"
            >
              <div>
                <label
                  className="x-checkbox-label x-body2"
                  htmlFor="610371595"
                >
                  <span
                    className="vh"
                  >
                    HBO
                     
                  </span>
                  <span
                    className="hg-price"
                  >
                    <span
                      className="hg-price__icon"
                    >
                      $
                    </span>
                    <span
                      className="hg-price__dollars"
                    >
                      19
                    </span>
                    <span
                      className="hg-price__cents"
                    >
                      .99
                    </span>
                    <span>
                      /mo
                    </span>
                  </span>
                </label>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
